xgi/algorithms/connected.py

```diff
@@ -214,35 +261,63 @@ def largest_connected_hypergraph(H, in_place=False):
     connected_components
     largest_connected_component
 
+    References
+    ----------
+    Aksoy, S.G., Joslyn, C., Ortiz Marrero, C. et al.
+    Hypernetwork science via high-order hypergraph walks.
+    EPJ Data Sci. 9, 16 (2020).
+    https://doi.org/10.1140/epjds/s13688-020-00231-0
+
     Example
     -------
     >>> import xgi
     >>> H = xgi.random_hypergraph(10, [0.1, 0.01], seed=1)
     >>> H_gcc = xgi.largest_connected_hypergraph(H)
     >>> print(H_gcc.num_nodes)
     6
-
     """
-    connected_nodes = max(connected_components(H), key=len)
+    connected_nodes = max(connected_components(H, s=s), key=len)
     if not in_place:
         return subhypergraph(H, nodes=connected_nodes).copy()
     else:
         H.remove_nodes_from(set(H.nodes).difference(connected_nodes))
 
 
-def _plain_bfs(H, source):
-    """A fast BFS node generator
+def _plain_bfs(H, source, s=1):
```

Comment:
maybe describe a little more (eg with a few inline comments) what the algo is doing